spec/spec.md

```diff
@@ -0,0 +1,180 @@
+-->
+
+# REUSE Specification
+
+This specification defines a standardized method for declaring copyright and
+licensing for software projects. The goal of the specification is to have
+unambiguous, machine-readable copyright and licensing information for each
+individual file in a project.
+
+## Definitions
+
+These are the definitions for some of the terms used in this specification:
+
+- Project --- ...
+
+- License file --- ...
+
+<!-- Also put a link to the SPDX Specification here somewhere -->
+
+<!-- Also DEP5 -->
+
+## License files
+
+A project must include a license file for every license under which files in the
+project are licensed.
+
+Each license file should be placed in the `LICENSES/` directory in the root of
+the project. The name of the license file should be the SPDX identifier of the
+license followed by a file extension (example: `GPL-3.0-or-later.txt`).
+
+<!-- FIXME: Rephrase "above recommendation". Maybe use paragraph numbers? -->
+
+A license file whose name does not conform to the above recommendation must
+contain, at the start of the file, one or more `Valid-License-Identifier` tags,
+followed by a `License-Text` tag, followed by the license text. Example:
+
+```
+Valid-License-Identifier: GPL-3.0-only
+Valid-License-Identifier: GPL-3.0-or-later
+License-Text:
+
+                    GNU GENERAL PUBLIC LICENSE
+                       Version 3, 29 June 2007
+
+ Copyright (C) 2007 Free Software Foundation, Inc. <https://fsf.org/>
+ Everyone is permitted to copy and distribute verbatim copies
+ of this license document, but changing it is not allowed.
+
+[...]
+```
+
+A project may contain a single license file outside of the `LICENSES/` directory
+in the file `COPYING`, `COPYRIGHT`, `LICENSE` or `LICENCE` in the root of the
+project.
+
+If a license does not exist in the SPDX License List, its SPDX identifier is
+`LicenseRef-[idstring]` as defined by the SPDX Specification.
+
+A project must not include license files for licenses under which none of the
+files in the project are licensed.
+
+Everything that applies to licenses in this section also applies to license
+exceptions, with the following exceptions:
+
+- Exception files must be placed in the `LICENSES/` directory in the root of the
+  project.
+
+- The tags `Valid-Exception-Identifier` and `Exception-Text` must be used
+  instead of `Valid-License-Identifier` and `License-Text`.
+
+<!-- Can we use LicenseRef for exceptions? -->
+
+<!-- TODO: Change "copyright and licensing information" to something shorter?
+Shorten it to CALI? -->
+
+## Copyright and licensing information
+
+Each file in the project must have copyright and licensing information
+associated with it, except the following files:
+
+- The license files.
+
+- The files belonging to the project's version control system (example:
+  `.git/`).
+
+- The files ignored by the version control system (example: `.gitignore`).
+
+- The files in the `.reuse/` directory in the root of the project.
```

Comment:
Apart from the DEP5 file, what else do we allow to be stored there?

Reply:
In the future, a tiny configuration file, and comment header templates.

Reply:
Then we should also put what goes into the `.reuse/` directory into the spec, just in case.

Reply:
@silverhook If I could predict the future, then yes :) I don't yet know exactly what the names of those files will be. I think this catch-all works sufficiently well. What would be the drawback of using this catch-all wording?

Reply:
@carmenbianca, well, let’s say you’re feeling pretty smart and smug and store code that you don’t want to have tools to find, but still plant in let’s say something you hold patents in.

How would you prevent such a malicious actor to

- simply store the patent-infested code into the `.reuse/` folder,
- not attach and FOSS license to that code
- in the rest of the code call code files from that folder, so that it pulls it in when building or running
- wait and laugh while the others make use of the trojan package

The package/repo will be REUSE compliant, according to REUSE Spec it will look FOSS as well. But as soon as  someone uses it or implements it into their own product/project, the malicious actor may strike with a patent lawsuit.

Reply:
@mxmehl Opinion?

@silverhook One way or another there is going to have to be a catch-all here, because the tool will (in the future) allow the user to put custom templates in there.

It could be half-fixed by saying something like "the configuration and template files in `.reuse/`", but this might force an update of the spec in case the tool ever decides to implement new stuff.

Reply:
You can always specify how the templates should look like and how their files have to be named.